File: Lexical.cpp
#include <string>
#include "Lexical.h"

bool isInt(std::string x) {
	//Functions that determines if a string is a valid int or not
	try {
		std::stoi(x);
		return true;
	}
	catch (...){
		return false;
	}
}

std::string getToken(std::string x) {
	//Takes in a string and detemines the token based off of it
	if ((x == "(") || (x == ")") || (x == "{") || (x == "}") || (x == ";") || (x == ",") || (x == "!") ||
		(x == "=") || (x == "+") || (x == "-") || (x == "*") || (x == "[") || (x == "]") || (x == "/")) {
		return ("'" + x + "'");
	}
	if (x == "void") return "T_Void";
	if (x == "func") return "T_Func";
	if (x == "int") return "T_Int";
	if (x == "string") return "T_String";
	if (x == "null") return "T_Null";
	if (x == "Print") return "T_Print";
	if (x == "return") return "T_Return";
	if ((x == "||") || (x == "&&")) return "T_logicaland";
	if (x == "if") return "T_If";
	if (x == "else") return "T_Else";
	if (x == "for") return "T_For";
	if (x == "while") return "T_While";
	if (x == "break") return "T_Break";
	if (x == "continue") return "T_Continue";
	if (x == "//") return "T_COMMENT";
	if (x == "==") return "T_Equivelent";
	if (x == "!=") return "T_NotEquivelent";
	if (x == "<") return "T_Less";
	if (x == "<=") return "T_LessEqual";
	if ((x[0] == '"') && (x[x.size() - 1] == '"')) return ("T_STRINGCONSTANT (value = " + x + ")");
	if (isInt(x)) return "T_IntConstant (value = " + x + ")";
	if ((x == "true") || (x == "false")) return "T_BoolConstant (value = " + x + ")";
	else {
		return "T_IDENTIFIER";
	}
}

void Lexical::setLexical(std::string x){
	lexical = x;
	token = getToken(x);
	//When functions call calls the getToken functions with given string
}

File: Lexical.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CS_4308_Term_Project
{
    class Lexical
    {
        public int line;
        public int startColumn;
        public int endColumn;
        public string token;
        public string lexical;
        public static bool isFloat(string str)
        {
            float result;
            return float.TryParse(str, out result);
        }

        static bool isInt(string str)
        {
            int result;
            return int.TryParse(str, out result);
        }


        public string getToken(string x)
        {
            //Takes in a string and detemines the token based off of it
            switch (x) {
                case "(":
                case ")":
                case "{":
                case "}":
                case ";":
                case ",":
                case "!":
                case "=":
                case "+":
                case "-":
                case "*":
                case "/":
                case "[":
                case "]":
                    return ("'" + x + "'");
                case "void":
                    return "T_Void";
                case "func":
                    return "T_Func";
                case "int":
                    return "T_Int";
                case "string":
                    return "T_String";
                case "float":
                    return "T_Float";
                case "null":
                    return "T_Null";
                case "Print":
                    return "T_Print";
                case "return":
                    return "T_Return";
                case "||":
                case "&&":
                    return "T_Logical";
                case "if":
                    return "T_If";
                case "else":
                    return "T_Else";
                case "for":
                    return "T_For";
                case "while":
                    return "T_While";
                case "break":
                    return "T_Break";
                case "continue":
                    return "T_Continue";
                case "//":
                    return "T_COMMENT";
                case "==":
                    return "T_Equivelent";
                case "!=":
                    return "T_NotEquivelent";
                case "<":
                    return "T_Less";
                case "<=":
                    return "T_LessEqual";
                case ">":
                    return "T_Greater";
                case ">=":
                    return "T_GreaterEqual";
                case "bool":
                    return "T_Bool";

                default:
                    if ((x[0] == '"') && (x[x.Length - 1] == '"')) return ("T_STRINGCONSTANT");
                    if (isInt(x)) return "T_IntConstant";
                    if (isFloat(x)) return "T_FloatConstant";
                    if ((x == "true") || (x == "false")) return "T_BoolConstant";
                    else
                    {
                        return "T_IDENTIFIER";
                    }
            }
        }

        public void setLexical(string x){
            lexical = x;
            token = getToken(x);
            //When functions call calls the getToken functions with given string
        }
    }
}